scripts/deploy/execute-deployment-pipeline.mjs

```diff
@@ -18,47 +18,55 @@ import shopifyJson from '../../packages/shopify/package.json' with {
 
 function getVersionComposants(version) {
   const parsedVersion = parse(version);
-  return {
-    major: parsedVersion?.major,
-    minor: parsedVersion?.minor,
-    patch: parsedVersion?.patch,
-    build: parsedVersion.prerelease[0],
-  };
+  return [
+    parsedVersion?.major,
+    parsedVersion?.minor,
+    parsedVersion?.patch,
+    parsedVersion.prerelease[0] || undefined,
+  ];
 }
+
+function getResolveVariableString(version, packageName) {
+  const prNumber = process.env.PR_NUMBER;
+  const versionComposantsOrdered = getVersionComposants(version);
+
+  // Use PR number as build if available and PATCH_ONLY is set
+  const {major, minor, patch} =
+    process.env.PATCH_ONLY && prNumber
```

Comment:
The PR Number is the "new" thing here, the rest is refactors/simplification (i.e. same behavior is expected)